### DISCUSSION TITLE
zero z offset after homing setup and mesh probe

### DISCUSSION THREAD

**Discussion by jxer** [Jyers/Marlin#781] [General Discussion]
is there a way to get z offset back to 0 in the main screen after, setting the inital offset with the bltouch, mine is a -3.49 or something, i would like this to be zero for fine tuning actual print runs, i feel like this can be saved in a setup menu, then zero'd out for normal prints, so its easier to see how much compensation i am adding between filaments etc....

**Comment by Tuncay-Ayhan** [comment]
What do you gain by changing it to 0.00 ? You can still fine-tune your print while printing. The only thing that remains is that you have to remember it. But if you are that concern/precise with your prints, you should fine-tune every print by changing the values by eye. The previous value is then invalid imho.

**Reply by jxer** [reply]
I change filament types constantly. they require different squish. you should be able to zero out like a caliper. no need to remember anything, and z offset should not ever change if using UBL

**Reply by Tuncay-Ayhan** [reply]
> I change filament types constantly. they require different squish. you should be able to zero out like a caliper. no need to remember anything, and z offset should not ever change if using UBL


True. You mean like:

Current z-offset = -3.49
Buttt let it pretend it's 0.00 while printing.

**Filament A** requires +0.02
**Filament B** requires -0.01

So you memorize that. And adjust accordingly. Is that what you mean?

**Reply by jxer** [reply]
Yes, have it set in the leveling  / home menu. Its not like i ever need to know it outside of setting the offset, where i would only be in that menu anyways.

then when printing have it 0 out, so you can make on the fly z changes with small numbers like .02 etc...

**Comment by jxer** [comment]
I guess not a lot of interest in this
